AsyncDisplayKitTests/ASTextKitTruncationTests.mm

```diff
@@ -0,0 +1,146 @@
+/* Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <UIKit/UIKit.h>
+#import <XCTest/XCTest.h>
+
+#import "ASTextKitContext.h"
+#import "ASTextKitTailTruncater.h"
+
+@interface ASTextKitTruncationTests : XCTestCase
+
+@end
+
+@implementation ASTextKitTruncationTests
+
+- (NSString *)_sentenceString
```

Comment:
We should surely add tests with CJK, Thai, Arabic, Cyrillic, Mongolian, Hebrew, etc :) - or at least one string with a lot of random shit in it :-P